apps/allocations/app/components/Panel/NewBudget.js

```diff
@@ -0,0 +1,116 @@
+import PropTypes from 'prop-types'
+import React from 'react'
+import { Info, TextInput } from '@aragon/ui'
+import { BigNumber } from 'bignumber.js'
+
+import { Form, FormField } from '../Form'
+import { isStringEmpty } from '../../utils/helpers'
+
+// TODO:: This should be votingTokens from account?
+const INITIAL_STATE = {
+  name: '',
+  nameError: true,
+  amount: '',
+  amountError: true,
+}
+
+class NewBudget extends React.Component {
+  static propTypes = {
+    onCreateBudget: PropTypes.func.isRequired
+  }
+
+  state =  INITIAL_STATE
+
+  changeField = e => {
+    if (e.target.name === 'name') {
+      if (isStringEmpty(e.target.value)) {
+        this.setState({ nameError: true })
+      }
+      else {
+        this.setState({ nameError: false })
+      }
+    }
+    else if (e.target.name === 'amount') {
+      if (isStringEmpty(e.target.value) || BigNumber(e.target.value).isNaN()) {
+        this.setState({ amountError: true })
```

Comment:
Sure, sorry for the confusion. I'm following the same validation methodology, namely

> Input validations should occur on the keystroke that causes them instead of  on submissions: e.g. if a user attempts to enter an allocation amount  for more ETH or Tokens than is available, the error message should show  up immediately and the submit button should become inactive

So, instead of waiting for the user to click on the form button to validate the inputs, they are validated on every keystroke.